fedora-builder/src/test/java/org/dataconservancy/nihms/builder/fs/DepositFileLabelMaketTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018 Johns Hopkins University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataconservancy.nihms.builder.fs;
+
+import org.dataconservancy.nihms.model.DepositFileType;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class DepositFileLabelMaketTest {
+
+
+    @Test
+    public void labelTest() {
+        DepositFileLabelMaker underTest = new DepositFileLabelMaker();
+
+        String label = underTest.label(DepositFileType.figure, null);
+        assertEquals("figure-1", label);
+
+        label = underTest.label(DepositFileType.figure, null);
+        assertEquals("figure-2", label);
+
+        label = underTest.label(DepositFileType.figure, "   ");
+        assertEquals("figure-3", label);
+
+        label = underTest.label(DepositFileType.supplemental, "figure-1");
+        assertEquals("figure-1", label);
+
+        label = underTest.label(DepositFileType.manuscript, "Moo Cows in the Pasture");
+        assertEquals("", label);
+
+        label = underTest.label(DepositFileType.figure, "Spotted Cows");
+        assertEquals("Spotted Cows", label);
```

Comment:
@jrmartino In my attempt to understand the logic of creating labels:

* the creation of a label for a manuscript with a non-empty description results in a zero-length label
* the creation of a label for a figure with a non-empty description results in the use of the description as a label?

Reply:
the specification requires labels for figure, table and supplemental. as the code stands now, these are the only labels that are created (can be easily changed). any other type will have a zero-length label.

if a required type file has a description which is not all white-space, the de-tabifeid and trimmed version of this description will be the candidate for the label. if this candidate is already used, we will add "-I" to it, where i is the smallest positive integer needed to make a new, unused label. 

this is the behavior i discussed with hanh - we can go over it an easily add types to the labeledTypes array to include more stuff. 

the critical type here is the supplemental type, whose label is used in creating hyperlinks

Reply:
the goal was to get a process which satisfies the spec - we have done so minimally